cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -283,7 +285,7 @@ func (c *Client) GetMemberID(peerURL string) (uint64, error) {
 			return member.GetID(), nil
 		}
 	}
-	return 0, nil
+	return 0, ErrNoMemberIDForPeerURL
```

Comment:
this seems valid, but i wonder if it will break something. are we running GetMemberID() in more places where we expect no error? if yes, we should probably adapt all code around it.

Reply:
Thanks for your review! This is a great question.

The only other call site is https://github.com/kubernetes/kubernetes/blob/65e306d7d85dcfcbb9e3f5e7373f5469cc4437c7/cmd/kubeadm/app/phases/etcd/local.go#L180.

I've mentally walked this code.

Previous to this change, if `GetMemberID` did not find a member ID for the peer, it returned `0`. Afterward, `MemberPromote` logged the message `[etcd] Member 0 already promoted.` The function continued to run, possibly returning an error later from the call to `WaitForClusterAvailable`.

With this change, if `GetMemberID` does not find the member ID, it returns an error, and that error is immediately returned.

Reply:
ack

Reply:
This looks valid.

Only the learner mode will use it.
https://github.com/kubernetes/kubernetes/blob/28247d53d22bbea9344305ac3db53c473c6d6278/cmd/kubeadm/app/phases/etcd/local.go#L175-L184

The change is OK if the GetMemberID returns the error, it is not valid and should return err. Or the promote will fail bellow.